### DISCUSSION TITLE
I love that the OnRecordAfterCreateRequest hook block execution of record creation

### DISCUSSION THREAD

**Discussion by SKumarSpace** [pocketbase/pocketbase#4364] [Show and tell]
I have a record that is being created where I need to do some post processing that requires me to call a third party API which may take some time. I love that `app.OnRecordAfterCreateRequest()` blocks until the request is finished.

Elegant design!

**Comment by luishugoooo** [comment]
For everyone from google: I fixed this by wrapping my function in a GoRoutine like this:

```
app.OnRecordAfterCreateSuccess("orders").BindFunc(func(e *core.RecordEvent) error {
		// Run mail sending in a goroutine to prevent blocking
		go func() {
			if err := mail.SendOrderPending(e); err != nil {
				log.Printf("Error sending order pending email: %v", err)
			}
		}()

		return e.Next()
	})
```

I don't know if this is recommended, but works for me.

**Reply by ganigeorgiev** [reply]
@FantixX It is fine if you want the email send to run in the background (_note that `e` is not concurrent safe for modifications in case you have another `OnRecordAfterCreateSuccess` hook that modifies the event fields_).

Alternatively you can also use the [`routine.FireAndForget(func() { ... })` helper](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.25.9/tools/routine#FireAndForget) as it comes with auto panic-recover handling. For the logging you can also check https://pocketbase.io/docs/go-logging/.
